### DISCUSSION TITLE
Snes reading black screen

### DISCUSSION THREAD

**Discussion by VamPikmin** [sanni/cartreader#1042] [Need help]
I've tried to build 6in1 v5, but since i'm mainly interested in Snes i only soldered it and the gb/a cart
 
Managed to dump a gba game but snes reading stays on black screen

Is it required to solder the nes and famicom cart connectors for the snes to work? 

Thank you for your help

**Comment by sanni** [comment]
No you don't have to solder the other slots. A black screen usually means it hangs when searching through the database on the SD card or issues with the clock generator.

For solving a possible SD issue you can try another SD card or format your current SD card with a program like "SD Card Formatter" or "Rufus".
For solving a possible issue with the clock generator you can disable the clock generator in Config.h by removing or uncommenting this line: https://github.com/sanni/cartreader/blob/master/Cart_Reader/Config.h#L55

**Comment by VamPikmin** [comment]
Thank you for your response.
I have tried two cards, but like you suggested reformatted and the issue persists.

I commented out the clockgen line and uploaded the sketch however still getting clock generator not found. So I removed the line completely and again same error

**Reply by VamPikmin** [reply]
I had to also uncomment line 555 enable_clockgen
I was able to dump a cart so the issue might be the clock gen.
I'll see if there's anything to find in the wiki
So far tested the points seems good

Thanks

**Reply by VamPikmin** [reply]
I enabled calibrate clock and get 
Read correction 0
😢

**Comment by handles888** [comment]
Chances are the clock generator is in upside down.  

When it’s in the sanni, the components should be facing the ceiling, not the bottom of the Sanni.

Go back to the instructions and make sure you didn’t solder the pins in so the components are facing the floor.   If its upside down, super Nintendo, Famicom and NES won’t work, but the others will.

**Reply by VamPikmin** [reply]
Thanks for your reply. I checked and it's not upside down, the components are on top and the pin legs on the bottom. I measured ~4.8V on vcc line when powered on. Using a phone charger

**Reply by handles888** [reply]
Hmmm.

This behavior is usually indicative of something wrong with the clock gen hardware

Might be defective. Try a replacement.

**Reply by VamPikmin** [reply]
Thank you
I don't have a spare, will order another maybe a different seller.
Does anyone have a link to aliexp for a working one?

Nothing i can do to troubleshoot the existing one?

**Reply by handles888** [reply]
Tenstar robot Great Wall 

I’ve had great luck with that Ali store.

You can pay a bit more and get a few on amazon quickly.  

I’ve only ever had one bad clock gen years back so I just replaced it.  Never tried to troubleshoot one, but I would absolutely give it a go if I got another bad one these days.  I’m much better at fixing broken electronics than I used to be.

I’d say check the voltage but you did that. Check for shorts on capacitors check for shorts on chip legs next to each other.  Check resistor values.

**Reply by VamPikmin** [reply]
Cheers for that, i'll order some

**Comment by VamPikmin** [comment]
I wanted to let you know the Tenstar robot clock gen worked. No error
I am getting an error with SA1 carts but it might be related to cic so I will open a separate thread
Thanks for your help
